### PR TITLE
test: add unit test for tree helpers

### PR DESCRIPTION
**Pull request by LucasSantos27**
**Increase to 49%**

Add unit tests for treeDetails and trees helper functions

- part of #1496

### CODE REVIEW

File: backend/kernelCI_app/tests/unitTests/helpers/treeDetails_test.py [hunk 1]
```diff
@@ -0,0 +1,897 @@
...
+    current_row_with_fail_status,
+    base_row_data,
+    row_data_with_unknown_compatible,
+    build_only_row_data,
+    test_only_row_data,
+    combined_row_data,
+)
+
+
+class TestCreateCheckoutsWhereClauses:
```
Comment (MarceloRobert) [inline backend/kernelCI_app/tests/unitTests/helpers/treeDetails_test.py:39]: Though the function is kinda confusing, this makes it make more sense to me. Could you just create another test for `name=something, branch=None`? That's a missing corner case

File: backend/kernelCI_app/tests/unitTests/helpers/treeDetails_test.py [hunk 2]
```diff
@@ -0,0 +1,897 @@
...
+        process_tests_issue(instance=instance, row_data=row_data, is_boot=True)
+
+        assert instance.failed_boots_with_unknown_issues == 1
+
+
+class TestDecideIfIsBuildFilteredOut:
+    def test_decide_if_is_build_filtered_out(self):
+        """Test decide_if_is_build_filtered_out function."""
+        instance = MagicMock()
+        instance.filters.is_build_filtered_out.return_value = True
```
Comment (MarceloRobert) [inline backend/kernelCI_app/tests/unitTests/helpers/treeDetails_test.py:545]: `decide_if_is_build_filtered_out` --> `MagicMock().filters.is_build_filtered_out.return_value = True`, I guess it's a simple check 😅 

I get the reason though, you can keep it until we decide to refactor those functions

File: backend/kernelCI_app/tests/unitTests/helpers/treeDetails_test.py [hunk 3]
```diff
@@ -0,0 +1,897 @@
...
+        assert instance.testStatusSummary["FAIL"] == 1
+        assert "x86_64-gcc" in instance.test_arch_summary
+        assert "defconfig" in instance.test_configs
+        assert "x86_64" in instance.testPlatformsWithErrors
+        assert instance.testFailReasons["Test error"] == 1
+        assert instance.testEnvironmentCompatible["hardware1"]["FAIL"] == 1
+
+    @patch("kernelCI_app.helpers.treeDetails.is_status_failure")
+    def test_process_test_summary_with_unknown_compatible(self, mock_is_status_failure):
+        """Test process_test_summary with unknown compatible environment."""
```
Comment (MarceloRobert) [inline backend/kernelCI_app/tests/unitTests/helpers/treeDetails_test.py:723]: Can you add a bit more to the comment of the `with_unknown_compatible` tests? I got a little confused about what was the difference between this and the base tests but then I got that the change is between `environment_compatible` and `platform`

Also for the boot `with_unknown_compatible` test

**Review by MarceloRobert — COMMENTED** [review]
Looks good, just some small comments.

What I'm getting from these unit test PRs is that we should have started adding descriptions to the functions much earlier... takes a bit of time to understand what each function does before checking their tests

**Comment by LucasSantos27** [comment]
> What I'm getting from these unit test PRs is that we should have started adding descriptions to the functions much earlier... takes a bit of time to understand what each function does before checking their tests

@MarceloRobert  We could add a criterion for accepting descriptions of the functions in addition to the tests.

**Comment by MarceloRobert** [comment]
>  We could add a criterion for accepting descriptions of the functions in addition to the tests.
@LucasSantos27 

You mean just checking manually or adding some kind of lint rule for this?

**Comment by MarceloRobert** [comment]
Either way, checking manually whenever someone makes a new PR sounds like a good start

**Comment by LucasSantos27** [comment]
> You mean just checking manually or adding some kind of lint rule for this?

I was thinking manually. I can add a new lint rule later. Nice Idea!

File: backend/kernelCI_app/tests/unitTests/helpers/fixtures/tree_details_data.py [hunk 4]
```diff
@@ -0,0 +1,181 @@
+from kernelCI_app.constants.general import UNKNOWN_STRING
+
+
+def create_row(**overrides):
+    """Create minimal row data with only necessary fields."""
```
[the remark this answers is not in the document]
Comment (LucasSantos27) [reply backend/kernelCI_app/tests/unitTests/helpers/fixtures/tree_details_data.py:5]: nice catch!
